### ISSUE TITLE
SOL token fetching change and fallback

### ISSUE THREAD

**Issue by MiroslavProchazka** [trezor/trezor-suite#10912]
As more coins and tokens are added to the Suite, we will experience an increased demand on Coingecko's requests limit. Currently, the limit is nearing its full capacity due to the implementation of Solana.

Therefore, it is necessary to modify the method by which we retrieve metadata for SOL and its tokens, specifically their names and fiat rates.

**Prerequisites:**
Currently, we are using an outdated JSON token list to fetch token names. This list is 2 years old and does not include all SOL tokens currently available. Additionally, fiat rates are being fetched from Coingecko.

The JSON needs to be updated, both initially and on a regular basis, to ensure that we have up-to-date data. You can find a dedicated script for updating the JSON [here](https://github.com/viaprotocol/tokenlists/blob/main/aggregate_tokens.py)

2) Then, we would put the adjusted JSON on our server ([trezor.io](http://trezor.io/)) and use it as the primary source for fetching token names. Coingecko requests needs to be added as a fallback in case something goes wrong with the list. However, we would still use Coingecko as the primary source for fiat rates.

3) This needs to be covered with [PR](https://github.com/trezor/trezor-suite/pull/10663) that is reducing the number of requests to the server.

4)Optionally, there is also space to reduce data from requests to the server - we are currently fetching information about the whitepaper and other unnecessary parameters..

5) In the end this should be covered by SOL definitions, but that is already part of another issue.

**Comment by gabrielKerekes** [comment]
We've "revived" the token list repository - https://github.com/nufi-official/tokenlists. It is being periodically updated again. Replacing the token list should as simple as switching [the url here](https://github.com/trezor/trezor-suite/blob/4c9abe1afcea26e98fa46675169152b847af71e1/packages/blockchain-link-utils/src/solana.ts#L34). Ideally though you would create a proxy under trezor.io which would point to the token list and reference the proxy in code.

**Comment by bosomt** [comment]
**QA OK**


**Info:**
 - Suite version: web 24.5.0 (34833c4751a901b72d446952fae03bff923423ac)
 - Browser: Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:125.0) Gecko/20100101 Firefox/125.0
 - OS: MacIntel
 - Screen: 2144x1206
 - Device: Trezor T2B1 2.7.0 regular (revision 45e8a842a31e62a6d43d7f6ccac62a45e1198ef0)
 - Transport: BridgeTransport 2.0.33

**Comment by Zaidalamari** [comment]
> As more coins and tokens are added to the Suite, we will experience an increased demand on Coingecko's requests limit. Currently, the limit is nearing its full capacity due to the implementation of Solana.
> 
> Therefore, it is necessary to modify the method by which we retrieve metadata for SOL and its tokens, specifically their names and fiat rates.
> 
> **Prerequisites:** Currently, we are using an outdated JSON token list to fetch token names. This list is 2 years old and does not include all SOL tokens currently available. Additionally, fiat rates are being fetched from Coingecko.
> 
> The JSON needs to be updated, both initially and on a regular basis, to ensure that we have up-to-date data. You can find a dedicated script for updating the JSON [here](https://github.com/viaprotocol/tokenlists/blob/main/aggregate_tokens.py)
> 
> 2. Then, we would put the adjusted JSON on our server ([trezor.io](http://trezor.io/)) and use it as the primary source for fetching token names. Coingecko requests needs to be added as a fallback in case something goes wrong with the list. However, we would still use Coingecko as the primary source for fiat rates.
> 3. This needs to be covered with [PR](https://github.com/trezor/trezor-suite/pull/10663) that is reducing the number of requests to the server.
> 
> 4)Optionally, there is also space to reduce data from requests to the server - we are currently fetching information about the whitepaper and other unnecessary parameters..
> 
> 5. In the end this should be covered by SOL definitions, but that is already part of another issue.
